test/util/mockgen.js

```diff
@@ -66,36 +63,143 @@ class Span {
+const mockSpan = () => {
+  return new Span()
+}
+
+let openApiMockGenerator
+
+// Factory generator for OpenApiRequestGenerator singleton
+const init = async () => {
+  if (!openApiMockGenerator) {
+    openApiMockGenerator = new OpenApiMockGenerator()
+    await openApiMockGenerator.load('./src/interface/openapi.yaml')
+  }
+  return openApiMockGenerator
+}
+
+const generateRequestHeaders = async (path, httpMethod, overrideRefs = null) => {
+  const generator = await init()
+  // Default header override refs
+  const defaultHeaderRefs = [
+    {
+      id: 'content-type',
+      pattern: 'application/vnd\\.interoperability\\.authorizations\\+json;version=1\\.0'
+    },
+    {
+      id: 'accept',
+      pattern: 'application/vnd\\.interoperability\\.authorizations\\+json;version=1\\.0'
+    },
+    {
+      id: 'date',
+      pattern: `${new Date().toUTCString()}`
+    }
+  ]
+
+  let localOverrideRefs
+  if (overrideRefs == null) {
+    localOverrideRefs = [...defaultHeaderRefs]
+  } else {
+    localOverrideRefs = [...overrideRefs]
+  }
+
+  const headers = await generator.generateRequestHeaders(path, httpMethod, localOverrideRefs)
+  delete headers['content-length']
+  return headers
+}
+
+const generateRequestBody = async (path, httpMethod, overrideRefs = null) => {
+  const generator = await init()
+
+  let localOverrideRefs
+  if (overrideRefs == null) {
+    localOverrideRefs = []
+  } else {
+    localOverrideRefs = [...overrideRefs]
   }
+  const body = await generator.generateRequestBody(path, httpMethod, localOverrideRefs)
+  return body
+}
 
-  mockGen = SwagMock(apiPath)
+const generateRequestQueryParams = async (path, httpMethod, overrideRefs = null) => {
+  const generator = await init()
 
-  /**
-   * Add an async version of requests
-   */
-  mockGen.requestsAsync = async (path, operation) => {
-    return new Promise((resolve, reject) => {
-      mockGen.requests(
-        { path, operation },
-        (error, mock) => error ? reject(error) : resolve(mock)
-      )
-    })
+  let localOverrideRefs
+  if (overrideRefs == null) {
+    localOverrideRefs = []
+  } else {
+    localOverrideRefs = [...overrideRefs]
   }
 
-  return mockGen
+  const params = await generator.generateRequestQueryParams(path, httpMethod, localOverrideRefs)
+
+  const result = {
+    params,
+    toString: () => {
+      return Object.entries(result.params).reduce((acc, [k, v]) => {
+        if (acc === '?') {
+          return `${acc}${k}=${v}`
+        } else {
+          return `${acc}&${k}=${v}`
+        }
+      }, '?')
+    },
+    toURLEncodedString: () => {
+      return encodeURI(result.toString())
+    }
+  }
+  return result
 }
 
-const mockSpan = () => {
-  return new Span()
+const generateRequest = async (path, httpMethod, override = null) => {
+  const localOverride = {
+    headers: null,
+    request: null
+  }
+  if (override != null) {
+    if (override.headers != null) {
+      localOverride.headers = [...override.headers]
```

Comment:
should headers be an array?

Reply:
Yes it appears that a library down the line iterates through the array and transforms it into a dict.